src/map.cpp

```diff
@@ -475,6 +475,9 @@ void map::move_vehicle( vehicle &veh, const tripoint &dp, const tileray &facing
             if( coll.type == veh_coll_veh ) {
                 continue;
             }
+            if( static_cast<size_t>( coll.part ) > veh.parts.size() ) {
```

Comment:
You probably meant to compare via `>=` because this will still allow invalid indizes (equal to `size()`), which causes UB in the array operator below.

Reply:
Fixed, thanks.